src/plot.js

```diff
@@ -91,9 +90,32 @@ export function plot(options = {}) {
   autoScaleLabels(channelsByScale, scaleDescriptors, axes, dimensions, options);
   autoAxisTicks(scaleDescriptors, axes);
 
+  // Reinitialize; for deriving channels dependent on other channels.
+  const newByScale = new Set();
+  for (const [mark, state] of stateByMark) {
+    if (mark.reinitialize != null) {
+      const {facets, channels} = mark.reinitialize(state.data, state.facets, state.channels, scales);
```

Comment:
If we pass state.data to mark.reinitialize here, we also need to allow the initializer to return new data…